### PR TITLE
chore: linkage monitor GitHub Actions for Java repository template

### PR DESCRIPTION
**Pull request by suztomo**
@stephaniewang526 @chingor13 

This PR migrates the Java library repositories to use the versioned action through cloud-opensource-java repository. Previously the Linkage Monitor was not versioned and caused unexpected problem in past (https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1838). To have proper versioning of the tool, we have prepared a versioned Linkage Monitor through `GoogleCloudPlatform/cloud-opensource-java/linkage-monitor@v1-linkagemonitor`.

As we no longer need `curl` to download the uber JAR, we don't need `linkage-monitor.sh`, I put the Maven install command in ci.yaml.

Details of the background: [go/jdd-github-actions](http://go/jdd-github-actions).

CC: @elharo 

# What is Linkage Monitor

Linkage Monitor works as a presubmit check in a GitHub repository to prevent open source libraries in the Google Cloud Java orbit (GAX, google-http-java-client, gRPC, etc.) from releasing versions that would introduce new linkage errors.

More on https://github.com/GoogleCloudPlatform/cloud-opensource-java/wiki/Linkage-Monitor

# What did I test?

I tested this yaml file in java-spanner and it worked.

https://github.com/googleapis/java-spanner/pull/831/files#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddR57

### CODE REVIEW

File: synthtool/gcp/templates/java_library/.kokoro/linkage-monitor.sh [hunk 1]
```diff
@@ -1,46 +0,0 @@
...
-echo ${JOB_TYPE}
-
-# attempt to install 3 times with exponential backoff (starting with 10 seconds)
-retry_with_backoff 3 10 \
-  mvn install -B -V \
-    -DskipTests=true \
-    -Dclirr.skip=true \
-    -Denforcer.skip=true \
-    -Dmaven.javadoc.skip=true \
-    -Dgcloud.download.skip=true
```
Comment (suztomo) [inline synthtool/gcp/templates/java_library/.kokoro/linkage-monitor.sh:39]: This command is now part of ci.yaml

File: synthtool/gcp/templates/java_library/.kokoro/linkage-monitor.sh [hunk 2]
```diff
@@ -1,46 +0,0 @@
...
-
-# Kokoro job cloud-opensource-java/ubuntu/linkage-monitor-gcs creates this JAR
-JAR=linkage-monitor-latest-all-deps.jar
-curl -v -O "https://storage.googleapis.com/cloud-opensource-java-linkage-monitor/${JAR}"
-
-# Fails if there's new linkage errors compared with baseline
-java -jar ${JAR} com.google.cloud:libraries-bom
```
Comment (suztomo) [inline synthtool/gcp/templates/java_library/.kokoro/linkage-monitor.sh:46]: These commands are done at the new action specified at `GoogleCloudPlatform/cloud-opensource-java/linkage-monitor@v1-linkagemonitor`. https://github.com/GoogleCloudPlatform/cloud-opensource-java/commit/a7f4d6322466b3579abdd4422d731160630653b9#diff-83a64340a0cc2b591f154efe6cc2f4928d0391788f54d76966bde9b7cb12d36cR10 (See the curl command downloads the versioned uber JAR)

**Comment by suztomo** [comment]
@Neenu1995 I see you got assigned to this. Let me know if you need any clarification.

**Review by chingor13 — APPROVED** [review]
Can you send a PR to any one of the libraries to test it out? Then we'll merge this one if it passes.

**Comment by chingor13** [comment]
> Can you send a PR to any one of the libraries to test it out? Then we'll merge this one if it passes.

Oh I see it was done in spanner.

**Comment by suztomo** [comment]
Thank you.

**Pull request merged**